--- a/net/minecraft/world/level/block/SlimeBlock.java
+++ b/net/minecraft/world/level/block/SlimeBlock.java
@@ -42,6 +_,7 @@
         Vec3 movement = entity.getDeltaMovement();
         if (movement.y < 0.0) {
             double factor = entity instanceof LivingEntity ? 1.0 : 0.8;
+            if (org.bxteam.divinemc.config.DivineConfig.FixesCategory.fixIncorrectBounceLogic) entity.setOnGround(movement.y > -0.15); // DivineMC - Carpet-Fixes: Fix Slime Block Bounce Logic
             entity.setDeltaMovement(movement.x, -movement.y * factor, movement.z);
         }
     }
